postile/postile.py

```diff
@@ -188,6 +197,30 @@ def prepared_query(filename):
         headers={"Content-Type": "application/x-protobuf"}
     )
 
+def preview(request):
+    """build and return a preview page
+    """
+    if app.debug:
+        template = jinja_env.get_template('index-debug.html')
+    else:
+        template = jinja_env.get_template('index.html')
+
+    html_content = template.render(host=request.host, scheme=request.scheme)
+    return response.html(html_content)
+
+def config_tm2(tm2file):
+    """Adds specific routes for tm2 source and prepare the global SQL Query
+
+    """
+    if not os.path.exists(tm2file):
+        print(f'file does not exists: {tm2file}')
+        sys.exit(1)
+    # build the SQL query for all layers found in TM2 file
+    Config.tm2query = prepared_query(tm2file)
+    # add route dedicated to tm2 queries
+    app.add_route(get_tile_tm2, r'/<z:int>/<x:int>/<y:int>.pbf', methods=['GET'])
+    app.add_route(preview, r'/', methods=['GET'])
```

Comment:
You add routes here in the case requests starts arriving before the app is configured, right?

I don't know how I feel about that: these requests would get a 404 then, and any of the 50x errors would be more appropriate (ideally 503).

I'd only annotate preview with a `@sanic.route` for now, and maybe in another PR, returns 503 if not ready, WDYT?

Reply:
> You add routes here in the case requests starts arriving before the app is configured, right?

No, i add routes before sanic starts because the function that responds to /z/x/y.pbf depends  on the command line invoked. 
`postile  --tm2` enters a different route than `postile` without this option (yes it's a bit magic but it will change in next PRs). And `/preview` is not available in case of a direct access to a postgis layer, so the decorator is not possible.

Reply:
Oh I see, thanks! One way to make this clearer would be to have a `config_route` instead of a config_tm2, and move the upper lever `if args.tm2` inside this function OR get rid of that extra `config_tm2` function alltogether, as it is only called once anyway.